video/vdu_stream_processor.h

```diff
@@ -188,7 +188,12 @@ void VDUStreamProcessor::processAllAvailable() {
 //
 void VDUStreamProcessor::processNext() {
 	if (byteAvailable()) {
-		vdu(readByte());
+		char ch = readByte();
+		#if ZDI==1 || SERIALKB == 1
+			// echo EZ80 characters to the debug host
+			DBGSerial.print (ch);
+		#endif
+		vdu(ch);
```

Comment:
echoing here to debug may be problematic.  this will be the raw unfiltered VDU stream, control characters and all.

you probably want to instead modify `vdu.h` where it's making a call to `plotCharacter(c)` and also `cursorCR()`